### PR TITLE
BugFix: NullableColumn should check the null_data when xor_checksum

### LINKED ISSUE #3438: checksum is not consistent between three replicas
The checksum algorithm(crc32) reads data rowsets one by one. There are some problems.
1. If there are three replicas A, B, C. The compaction progress is not consistent.
     If A has three rowset, B has two rowsets, C has one rowset. The checksum will be different.
2. The chunk size between different replicas are not the same. A may set the chunk_size to be 4096, B is 2048, C is 1024.
    Crc32 disobey associativity. So the checksum is not consistent between replicas.

### PR DESCRIPTION
**Pull request by chaoyli**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #3438

### CODE REVIEW

File: be/src/column/decimalv3_column.cpp [hunk 1]
```diff
@@ -82,36 +82,14 @@ void DecimalV3Column<T>::crc32_hash(uint32_t* hash, uint32_t from, uint32_t to)
...
-int64_t DecimalV3Column<T>::xor_checksum() const {
+int64_t DecimalV3Column<T>::xor_checksum(uint32_t from, uint32_t to) const {
     // The XOR of DecimalV3Column
     // XOR all the decimals one by one
     auto& data = this->get_data();
     int64_t xor_checksum = 0;
-    size_t num = data.size();
     const T* src = reinterpret_cast<const T*>(data.data());
 
-#ifdef __AVX2__
```
Comment (imay) [inline be/src/column/decimalv3_column.cpp:93]: why did you remove SIMD code?

File: be/src/column/nullable_column.cpp [hunk 2]
```diff
@@ -325,26 +325,11 @@ int64_t NullableColumn::xor_checksum() const {
...
-    }
-    uint8_t* checksum_vec = reinterpret_cast<uint8_t*>(&avx2_checksum);
-    for (size_t i = 0; i < step; ++i) {
-        xor_checksum ^= checksum_vec[i];
-    }
-#endif
-
     for (size_t i = 0; i < num; ++i) {
         xor_checksum ^= src[i];
+        xor_checksum ^= _data_column->xor_checksum(i, i + 1);
```
Comment (stdpain) [inline be/src/column/nullable_column.cpp:330]: I couldn't understand.
may be ?
```
if (!src[i]) xor_checksum ^= _data_column->xor_checksum(i, i + 1);
```

Comment (chaoyli) [reply be/src/column/nullable_column.cpp]: Fixed

File: be/src/column/decimalv3_column.cpp (hunk 1 above)
Comment (chaoyli) [reply be/src/column/decimalv3_column.cpp:93]: The compiler has already optimized the easy loop. And I test also it has no performance differenece.

**Review by stdpain — APPROVED** [review]

**Review by trueeyu — APPROVED** [review]
